src/styles/index.scss

```diff
@@ -1,3 +1,62 @@
-body {
+@import './utils/variables';
+
+* {
+  box-sizing: border-box;
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+  width: calc(
+    #{$size-of-cell} * #{$number-of-columns} + #{$gap} * 6 + #{$padding} * 2
+  );
+
+  &__day {
+    height: 100px;
+    width: 100px;
+    border: 1px solid #000;
+    background-color: $color-of-cell;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    transition: all $duration-of-animation;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $color-on-hover;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $days from 1 through 31 {
+    &__day:nth-child(#{$days})::before {
+      content: '#{$days}';
+    }
+  }
+
+  @each $day, $index in $days-of-week {
+    &--start-day-#{$day} .calendar__day:first-child {
+      margin-left: calc((#{$size-of-cell} + #{$gap}) * #{$index});
+    }
+  }
+
+  @for $last-days from 28 to 31 {
+    &--month-length-#{$last-days}
+      .calendar__day:nth-child(n + #{$last-days + 1}) {
+      display: none;
+    }
```

Comment:
The class naming convention for '--month-length-#{$last-days}' is not clear. Consider using a more descriptive naming that indicates the purpose of these modifiers, such as '--max-days-#{$last-days}' to indicate that it sets the maximum number of days in the calendar.